src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -354,36 +354,39 @@ public ReadPileup getOverlappingFragmentFilteredPileup() {
      *
      * @return the newly filtered pileup
      */
-    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, Comparator<PileupElement> tieBreaker) {
-        Map<String, PileupElement> filteredPileup = new HashMap<String, PileupElement>();
-        Set<String> readNamesDeleted = new HashSet<>();
+    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, Comparator<PileupElement> tieBreaker, SAMFileHeader header) {
+        List<PileupElement> filteredPileupList = new ArrayList<PileupElement>();
 
-        for (PileupElement p : this) {
-            String readName = p.getRead().getName();
+        for( Map.Entry<String, ReadPileup> pileup : this.splitBySample(header, null).entrySet()) {
+            Map<String, PileupElement> filteredPileup = new HashMap<String, PileupElement>();
```

Comment:
Done. I'm not particularly worried about speed here. We probably won't use this tool except with a single sample, and if it's a problem for users I'd propose that we can deal with it then.